src/bio2bel/sources/biogrid.py

```diff
@@ -0,0 +1,265 @@
+    'ID Interactor B': TARGET,
+    'Alt IDs Interactor A': ALT_SOURCE_ID,
+    'Alt IDs Interactor B': ALT_TARGET_ID,
+    'Interaction Types': RELATION,
+    'Publication Identifiers': PUBMED_ID,
+}
+
+log = logging.getLogger(__name__)
+
+
+def _get_my_df() -> pd.DataFrame:
+    """Get the BioGrid dataframe."""
+    path = ensure_path(prefix=MODULE_NAME, url=URL)[:-3] + 'txt'
+    log.info(path)
+    return pd.read_csv(path, sep='\t', dtype=str)
+
+
+def filter_for_prefix_single(list_ids: Iterable[str], prefix: str, rstrip: str = ' ', lstrip: str = ' ',
+                             separator: str = '|') -> List[List[str]]:
+    """Split the Iterable by the separator.
+
+    :param separator: separator between ids
+    :param prefix: prefix to filter for (e.g. 'pubmed')
+    :param rstrip: characters to strip from split value from left
+    :param lstrip: characters to strip from split value from right
+    :param list_ids: list of identifiers
+    :return: filtered list of ids
+    """
+    final_list = []
+    for ids in list_ids:
+        id_list = ids.split(separator)
+        flag = False
+        for i in id_list:
+            if i.startswith(prefix):
+                final_list.append(i.lstrip(lstrip).rstrip(rstrip))
+                flag = True
+        if not flag:
+            final_list.append('nan')
+    return final_list
+
+
+def filter_for_prefix_multi(list_ids: Iterable[str], prefix: str, separator: str = '|') -> List[List[str]]:
+    """Split the Iterable by the separator.
+
+    :param separator: separator between ids
+    :param prefix: prefix to filter for (e.g. 'pubmed')
+    :param list_ids: list of identifiers
+    :return: filtered list of lists of ids
+    """
+    final_list = []
+    for ids in list_ids:
+        id_list = ids.split(separator)
+        flag = False
+        row_list = []
+        for i in id_list:
+            if i.startswith(prefix):
+                row_list.append(i)
+                flag = True
+        # if no matching value with prefix is found, append 'nan' as individual value
+        if not flag:
+            final_list.append('nan')
+        # if at least one value is found, append list
+        if len(row_list) > 0:
+            final_list.append(row_list)
+    return final_list
+
+
+def expand_df(df: pd.DataFrame, column_name: str) -> pd.DataFrame:
+    """Add row with same content to dataframe if there are multiple values in one column.
+
+    The specified column must contain an Iterable.
+
+    :param column_name: name of column with multiple values to be simplified
+    :param df: dataframe with multiple values in one column
+    :return: dataframe with expanded values
+    """
+    values_data = df.values
+    column_index = df.columns.get_loc(column_name)
+    # index, row of each drug
+    for index, value_list in enumerate(tqdm(values_data, desc='Expanding dataframe ...')):
+        # if there are multiple entries in the column
+        if len(value_list[column_index]) > 1:
```

Comment:
why not just iterate in an inner loop in the actual function that works on the dataframe, instead of doing very complicated (and hard to understand) manipulations of the dataframe?

Reply:
also this code doesn't work...

Reply:
I found a simple way to achieve the expansion of the dataframe if there are multiple values in a cell: 47de9de5bfe4e77328058f3c5071870ec27b0756
Closed ;)